lib/ar_transaction_changes.rb

```diff
@@ -41,14 +41,25 @@ def _store_transaction_changed_attributes(attr_name)
     old_value = _read_attribute_for_transaction(attr_name)
     ret = yield
     new_value = _read_attribute_for_transaction(attr_name)
-    unless transaction_changed_attributes.key?(attr_name) || new_value == old_value
+    if !transaction_changed_attributes.key?(attr_name) && new_value != old_value
       attribute = @attributes[attr_name]
       transaction_changed_attributes[attr_name] = if attribute.type.is_a?(::ActiveRecord::Type::Serialized)
         attribute.type.deserialize(old_value)
       else
         old_value
       end
       transaction_changed_attributes
+    elsif transaction_changed_attributes.key?(attr_name)
+      attribute = @attributes[attr_name]
+      if attribute.type.is_a?(::ActiveRecord::Type::Serialized)
```

Comment:
Can you add a test for this?

Reply:
added

Reply:
The tests still pass with this conditional branch removed

Reply:
yes the test for the array (maybe even the code) does not work properly, looking into it

Reply:
Note that I think there is still a problem with handling in place mutations of serialized attributes, so avoid trying to test that for this PR.  There was a branch with a fix (https://github.com/dylanahsmith/ar_transaction_changes/compare/master...Shopify:use_changes_applied) but I think there was some blocker that was discovered when testing on Shopify Core CI.